src/testing/testing.go

```diff
@@ -0,0 +1,54 @@
+package testing
+
+import (
+	"fmt"
+)
+
+// T is a test helper.
+type T struct {
+}
+
+// TestToCall is a reference to a test that should be called during a test suite run.
+type TestToCall struct {
+	// Name of the test to call.
+	Name string
+	// Function reference to the test.
+	Func func(*T)
+}
+
+// M is a test suite.
+type M struct {
+	// tests is a list of the test names to execute
+	Tests []TestToCall
+}
+
+// Run the test suite.
+func (m *M) Run() int {
+	for _, test := range m.Tests {
+		t := &T{}
+		test.Func(t)
+	}
+	// TODO: detect failures and return a failing exit code
+	// Right now we can't handle one anyway so it doesn't matter much
+	return 0
+}
+
+// Fatal is equivalent to Log followed by FailNow
+func (t *T) Fatal(args ...string) {
+	// This doesn't print the same as in upstream go, but works good enough
+	fmt.Print(args)
+	//t.FailNow()
+}
+
+/*
+func (t *T) FailNow() {
+	// This fails with
+Undefined symbols for architecture x86_64:
+"_syscall.Exit", referenced from:
+  _main in main.o
+ld: symbol(s) not found for architecture x86_64
+clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Comment:
`syscall.Exit` is implemented in the runtime, as can be seen here:

https://github.com/golang/go/blob/cb8aefd3b06f39679ebe1abbd38abef52cbd76b7/src/syscall/syscall.go#L102-L105

The implementation is here:

https://github.com/golang/go/blob/50bd1c4d4eb4fac8ddeb5f063c099daccfb71b26/src/runtime/runtime.go#L61

It could be implemented in TinyGo in the following way in the runtime package:

```go
//go:linkname exit exit
func exit(code int32)

//go:linkname syscall_Exit syscall.Exit
func syscall_Exit(cide int) {
    exit(int32(code))
}
```

What this basically does is implement `syscall.Exit` by calling the libc `exit` function.
You could make a separate PR for this feature.

Reply:
Thanks!

Reply:
Reading it again, I see I made a mistake. The `exit` declaration should be the following:

```go
//go:export exit
func exit(code int32)
```

(`//go:export` is a bit of a misnomer, we're importing a function here. But it lets the compiler know it should follow the C ABI).

Reply:
I tried to get this to work, and while there aren't any errors, it doesn't emit the specified exit code either. Maybe someone else can pick this up and get it to work.

Reply:
Hmmm maybe it should be `//go:extern exit`?

Reply:
Oh wait! I wasn't doing it right. I was testing it with `tinygo run` but after trying it with `tinygo build` then running the binary it created, that worked. So let me try more to get it to work with `tinygo test`.

Reply:
So I thought it didn't work so I wrote my own: https://github.com/tinygo-org/tinygo/pull/380 (sorry, didn't see your comment, don't want to cause duplicate work...)
And had the exact same realization. We should probably fix this, it makes logical sense that `tinygo run` exits with the same exit code as the program you're trying to run. But that is somewhat of a separate issue.